src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,158 @@
+	/// <returns>Value indicating if the animation completed successfully or not</returns>
+	public static Task<bool> TextColorTo(this " + textStyleClass.Namespace + "." + textStyleClass.ClassName + @" element, Color color, uint rate = 16u, uint length = 250u, Easing? easing = null)
+	{
+		ArgumentNullException.ThrowIfNull(element);
+		ArgumentNullException.ThrowIfNull(color);
+
+		if(element is not Microsoft.Maui.ITextStyle)
+			throw new ArgumentException($""Element must implement {nameof(Microsoft.Maui.ITextStyle)}"", nameof(element));
+
+		//Although TextColor is defined as not-nullable, it CAN be null
+		//If null => set it to Transparent as Animation will crash on null BackgroundColor
+		element.TextColor ??= Colors.Transparent;
+
+		var animationCompletionSource = new TaskCompletionSource<bool>();
+
+		try
+		{
+			new Animation
+			{
+				{ 0, 1, GetRedTransformAnimation(element, color.Red) },
+				{ 0, 1, GetGreenTransformAnimation(element, color.Green) },
+				{ 0, 1, GetBlueTransformAnimation(element, color.Blue) },
+				{ 0, 1, GetAlphaTransformAnimation(element, color.Alpha) },
+			}
+			.Commit(element, nameof(TextColorTo), rate, length, easing, (d, b) => animationCompletionSource.SetResult(true));
+		}
+		catch (ArgumentException aex)
+		{
+			//When creating an Animation too early in the lifecycle of the Page, i.e. in the OnAppearing method,
+			//the Page might not have an 'IAnimationManager' yet, resulting in an ArgumentException.
+			System.Diagnostics.Debug.WriteLine($""{aex.GetType().Name} thrown in {typeof(ColorAnimationExtensions_" + textStyleClass.ClassName + @").FullName}: {aex.Message}"");
+			animationCompletionSource.SetResult(false);
+		}
+
+		return animationCompletionSource.Task;
+
+
+		static Animation GetRedTransformAnimation(" + textStyleClass.Namespace + "." + textStyleClass.ClassName + @"  element, float targetRed) =>
+			new(v => element.TextColor = element.TextColor.WithRed(v), element.TextColor.Red, targetRed);
+
+		static Animation GetGreenTransformAnimation(" + textStyleClass.Namespace + "." + textStyleClass.ClassName + @"  element, float targetGreen) =>
+			new(v => element.TextColor = element.TextColor.WithGreen(v), element.TextColor.Green, targetGreen);
+
+		static Animation GetBlueTransformAnimation(" + textStyleClass.Namespace + "." + textStyleClass.ClassName + @"  element, float targetBlue) =>
+			new(v => element.TextColor = element.TextColor.WithBlue(v), element.TextColor.Blue, targetBlue);
+
+		static Animation GetAlphaTransformAnimation(" + textStyleClass.Namespace + "." + textStyleClass.ClassName + @"  element, float targetAlpha) =>
+			new(v => element.TextColor = element.TextColor.WithAlpha(v), element.TextColor.Alpha, targetAlpha);
+	}
+}");
+			var source = textColorToBuilder.ToString();
+			SourceGeneratorHelpers.FormatText(ref source, options);
+			context.AddSource($"{textStyleClass.ClassName}TextColorTo.g.shared.cs", SourceText.From(source, Encoding.UTF8));
```

Comment:
Is there any reason for the `.shared` in the name?

I'm concerned about the compiler not considering this as auto generated file in some scenarios. See compiler implementation for reference.

https://github.com/dotnet/roslyn/blob/ffde321ee629f942e346318503766ea367393533/src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs#L72-L82

Reply:
It's fine, the file just enter that loop when his name doesn't end with `.cs`, this file ends with `.cs`, so this will compile (:

Reply:
@pictos It's not about "will compile" or not. The `AnalyzerDriver` in roslyn treats generated code differently per the registered [`GeneratedCodeAnalysisFlags`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.generatedcodeanalysisflags?view=roslyn-dotnet-4.0.1). So it's better to drop the `.shared` part IMO if there is no benefit/reason for it.